container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java

```diff
@@ -55,13 +58,18 @@ public SearchCluster(String clusterId, SearchGroupsImpl groups, VipStatus vipSta
 
     @Override
     public String name() { return clusterId; }
-    public VipStatus getVipStatus() { return vipStatus; }
+
+    public void updateNodes(Collection<Node> newNodes, double minActivedocsPercentage) {
+        Collection<Node> retainedNodes = groups.nodes();
+        Collection<Node> currentNodes = new HashSet<>(newNodes);
+        retainedNodes.retainAll(currentNodes);
+        currentNodes.removeIf(retainedNodes::contains);
+        currentNodes.addAll(retainedNodes);
```

Comment:
What does this do?

Reply:
Replaces the newly created, information-less nodes, with the ones that we have updated with information from the previous monitor.